resources/2024-07-secretstore-feature-spec.md

```diff
@@ -0,0 +1,426 @@
+    host: substring(cosmosAccount.properties.documentEndpoint, 0, lastIndexOf(cosmosAccount.properties.documentEndpoint, ':'))
+    port: int(split(substring(cosmosAccount.properties.documentEndpoint,lastIndexOf(cosmosAccount.properties.documentEndpoint, ':') + 1), '/')[0])
+    database: cosmosAccount::database.name
+    username: ''
+    resources: [
+      { id: cosmosAccount.id }
+    ]
+    secrets: {
++        connectionString: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'connectionString'
++            }
++          }
++        }
++        password: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'password'
++            }
++          }
++        }
+    }
+  }
+}
+```
+
+**In a messaging resource:**
+```diff
+resource rabbitmq 'Applications.Messaging/rabbitmqQueues@2023-10-01-preview' = {
+  name: 'rabbitmq'
+  properties: {
+    environment: environment
+    application: app.id
+    resourceProvisioning: 'manual'
+    queue: 'radius-queue'
+    host: rmqHost
+    port: rmqPort
+    vHost: vHost
+    username: rmqUsername
+    secrets: {
++      password: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'password'
++            }
++          }
++      }
+    }
+  }
+}
+```
+
+**In a Dapr resource:**
+```diff
+resource config 'Applications.Dapr/configurationStores@2023-10-01-preview' = {
+  name: 'configstore'
+  properties: {
+    environment: environment
+    application: app.id
+    resourceProvisioning: 'manual'
+    type: 'configuration.redis'
+    metadata: {
+      redisHost: {
+        value: '<REDIS-URL>'
+      }
+      redisPassword: {
++        value: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'password'
++            }
++          }
++        }
+      }
+    }
+    version: 'v1'
+  }
+}
+```
+> Note: The proposal here is to follow the same `valueFrom` syntax as the `Applications.Core/containers` resource type for referencing Radius Secrets in the `Applications.Dapr/*` resource type, which deviates from the [Dapr Secrets reference pattern](https://docs.radapp.io/guides/author-apps/dapr/how-to-dapr-secrets/) that exists today to reference Dapr Secret Stores. We'll leave it up to the technical design to decide whether to follow the existing Dapr Secrets reference pattern or to follow the `Applications.Core/containers` reference pattern for Radius Secret Store resources.
+
+**In a custom UDT resource:**
+```diff
+resource customtype 'Applications.Core/udt@2023-10-01-preview' = {
+  name: 'customtype'
+  properties: {
+    application: application
+    environment: environment
+    env:{
++        URI: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'uri'
++            }
++          }
++        }
+      }
```

Comment:
We generally expect env variables on container. How/where do envision this being used?